### PR TITLE
Improve builder profiles

### PR DESCRIPTION
**Pull request by vvaldez**
* Moves all common tasks to be variable driven
    * Loads profile variables depending on requested 'ami_config'
    * Loads profile-specific tasks based on requested 'ami_config'
    * Uses combination of default 'all_yum_packages' and profile-specific 'yum_packages'
    * Uses combination of default 'all_yum_repos' and profile-specific 'yum_repos'
    * Uses combination of default 'all_pip_packages' and profile-specific 'pip_packages'
    * Adds toggle variables for 'epel_setup', 'yum_update', and 'ami_create'
    * Removes install of boto packages and change all ec2 modules to run on localhost
    * New profiles can easily be added by copying existing structure using a file or directory

Run playbok as previous to generate an AMI. However, several toggles have been added. For example, to produce an AMI for Tower and no do a yum update:

```
ansible-playbook -e @vvaldez-secrets.yml ami_lab_launch.yml  -e ami_user=vvaldez -e ami_unique_id=$(( AMIVER += 1 )) -e ami_inst_type=t2.small -e ami_config=tower -e ami_create=True  -e yum_update=False -e epel_setup=True -v
```

To create an OCP AMI, run:
```
ansible-playbook -e @vvaldez-secrets.yml ami_lab_launch.yml  -e ami_user=vvaldez -e ami_unique_id=$(( AMIVER += 1 )) -e ami_inst_type=t2.medium -e ami_config=ocp -e ami_create=True -v
```

To create an Ansible Host but not snap an AMI:
```
ansible-playbook -e @vvaldez-secrets.yml ami_lab_launch.yml  -e ami_user=vvaldez -e ami_unique_id=$(( AMIVER += 1 )) -e ami_inst_type=t2.medium -e ami_config=ocp -e ami_create=False  -e yum_update=True -v
```

Note that some of these are defaults, I was just showing the variables. See roles/ami_config/vars/ for profile-specific variable defaults.

### CODE REVIEW

**Comment by scollier** [comment]
@tchughesiv just FYI @ this refactor.  Shouldn't change anything functionally.

**Comment by scollier** [comment]
@vvaldez please update https://github.com/scollier/ansible-ami-builder/blob/master/docs/getting-started.md with the PR install instructions?

**Comment by scollier** [comment]
Tested: ocp

```
/ TASK [ami_config : Create and attach a volume to the AMI so \
\ the Docker storage can be setup]                            /
 -------------------------------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

Thursday 19 April 2018  10:37:50 -0400 (0:00:00.097)       0:12:10.812 ******** 
Thursday 19 April 2018  10:37:50 -0400 (0:00:00.097)       0:12:10.811 ******** 
fatal: [ami-scollier-ocp-4.labs.sysdeseng.com]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'result' is undefined\n\nThe error appears to have been in '/home/scollier/git_projects/ansible-ami-build/roles/ami_config/tasks/profiles/ocp.yml': line 16, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Create and attach a volume to the AMI so the Docker storage can be setup\n  ^ here\n\nexception type: <class 'ansible.errors.AnsibleUndefinedVariable'>\nexception: 'result' is undefined"}
 ____________
```

debugging now.


Tried with:

```
ansible-playbook -e @scollier_secrets-ami_build.yml ami_lab_launch.yml  -e ami_user=scollier -e ami_unique_id=$(( AMIVER += 1 )) -e ami_inst_type=t2.small -e ami_config=ocp -e ami_create=True  -e yum_update=False
```

and

```
ansible-playbook -e @scollier_secrets-ami_build.yml ami_lab_launch.yml  -e ami_user=scollier -e ami_unique_id=$(( AMIVER += 1 )) -e ami_inst_type=t2.small -e ami_config=ocp -e ami_create=True  -e yum_update=False -e epel_setup=True
```

**Comment by scollier** [comment]
rh-container-lab AMI build worked.

**Comment by scollier** [comment]
tower AMI build worked.

File: roles/ami_config/tasks/profiles/ocp.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    var: ec2_instances.instances[0].instance_id
+    verbosity: 2
+
+- name: Create and attach a volume to the AMI so the Docker storage can be setup
+  local_action:
+    module: ec2_vol
+    aws_access_key: "{{ ec2_access_key }}"
+    aws_secret_key: "{{ ec2_secret_key }}"
+    region: "{{ aws_region }}"
+    instance: "{{ result.instances[0].instance_id }}"
```
Comment (scollier) [inline roles/ami_config/tasks/profiles/ocp.yml]: "result" should be "ec2_instances".  This was causing my breakage.

**Comment by scollier** [comment]
OCP AMI build worked after changing that result variable.

**Comment by vvaldez** [comment]
@scollier thanks for the review, doc updated as requested.

**Comment by scollier** [comment]
LGTM

**Pull request merged**